src/mqttcm_connect.c

```diff
@@ -1573,6 +1596,87 @@ int isSubscribeNeeded(char *compname)
 	return 2;
 }
 
+char* GetTopicFromSubcribeId(int subscribeId)
+{
+	comp_topic_name_t* temp = g_head;
+	while (temp != NULL)
+	{
+		if(temp->subscribeId == subscribeId)
+		{
+			return (temp->compName);
+		}
+		temp = temp->next;
+	}
+	return NULL;
+}
+
+void AddSubscriptionIdToList(char *comp, int subscribeId)
+{
+	comp_topic_name_t* temp = g_head;
+	while (temp != NULL)
+	{
+		if(strcmp(temp->compName, comp) == 0)
+		{
+			temp->subscribeId = subscribeId;
+			MqttCMInfo("The component %s is subscribed to topic %s with subscribeId %d\n", temp->compName, temp->topic, subscribeId);
+			return;
+		}
+		temp = temp->next;
+	}
+}
+
+void insert(char * compName, char* topic)
+{
+
+	if( (compName != NULL) && (topic != NULL) )
+	{
+		comp_topic_name_t* newNode = (comp_topic_name_t*)malloc(sizeof(comp_topic_name_t));
+		if(newNode)
+		{
+			memset(newNode, 0, sizeof(comp_topic_name_t) );
+			strncpy(newNode->compName, compName, sizeof(newNode->compName) - 1);
+			strncpy(newNode->topic, topic, sizeof(newNode->topic) - 1);
+			newNode->subscribeOnFlag = 0;
```

Comment:
If compName and topic are added to list, then we need to set newNode->subscribeOnFlag to 1 here right ? what is the use case for subscribeOnFlag, add it in a comment